onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -252,9 +252,12 @@ class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, Ga
 class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 7, 9, float, Gemm);
 class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 7, 9, double, Gemm);
 class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 7, 9, MLFloat16, Gemm);
-class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, float, MatMul);
-class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, double, MatMul);
-class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, MLFloat16, MatMul);
+class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, 8, float, MatMul);
+class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, 8, double, MatMul);
+class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, 8, MLFloat16, MatMul);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 9, float, MatMul);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 9, double, MatMul);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 9, MLFloat16, MatMul);
```

Comment:
>MatMul [](start = 95, length = 6)

opset 9 added type int32, int64, do we need to support it or just fallback to CPU?

Reply:
cublas has no support for int32 x int32 -> int32, same for int64. Those will fallback to whatever EPs that support them.

---
In reply to: [287527378](https://github.com/microsoft/onnxruntime/pull/1106#discussion_r287527378) [](ancestors = 287527378)

Reply:
should we cast it to float then cast back? or you think it won't help for performance and accuracy?

---
In reply to: [287529390](https://github.com/microsoft/onnxruntime/pull/1106#discussion_r287529390) [](ancestors = 287529390,287527378)

Reply:
accuracy and performance would be the concern. We can work on it later if needed

---
In reply to: [287530031](https://github.com/microsoft/onnxruntime/pull/1106#discussion_r287530031) [](ancestors = 287530031,287529390,287527378)

Reply:
I think Tianlei is working on that for the cast part.

---
In reply to: [287531929](https://github.com/microsoft/onnxruntime/pull/1106#discussion_r287531929) [](ancestors = 287531929,287530031,287529390,287527378)